docker-compose-separate-containers.yml

```diff
@@ -1,23 +1,25 @@
-version: '2'
+version: '3'
 services:
   nginx:
     image: nginx
     container_name: nginx
     ports:
       - "80:80"
     volumes:
-      - /etc/nginx/conf.d
+      - nginx_conf:/etc/nginx/conf.d
 
   dockergen:
     image: jwilder/docker-gen
     command: -notify-sighup nginx -watch /etc/docker-gen/templates/nginx.tmpl /etc/nginx/conf.d/default.conf
-    volumes_from:
-      - nginx
     volumes:
       - /var/run/docker.sock:/tmp/docker.sock:ro
       - ./nginx.tmpl:/etc/docker-gen/templates/nginx.tmpl
+      - nginx_conf:/etc/nginx/conf.d
```

Comment:
Did you make sure to start the `dockergen` service before the `nginx` service? Because startup order is not guaranteed, and that's why the problem could happen sometimes.